cns/NetworkContainerContract.go

```diff
@@ -220,3 +227,29 @@ type UnpublishNetworkContainerResponse struct {
 	UnpublishStatusCode   int
 	UnpublishResponseBody []byte
 }
+
+// Validate - Validates network container request policies
+func (networkContainerRequestPolicy *NetworkContainerRequestPolicies) Validate() error {
+
+	// validate ACL policy
+	if strings.EqualFold(networkContainerRequestPolicy.Type, "ACLPolicy") {
+
+		var requestedAclPolicy hcn.AclPolicySetting
```

Comment:
This is specific to hnsv2 ( hcn.AclPolicySetting ). This may not work if the client is using docker and is specifying acl policy which is hnsv1 specific. This is not an issue at the moment but I think we should restrict the validation for EndpointType with APIPA for now. i.e. currently we only support this for APIPA endpoint type.
We can discuss more on this when we want to extend this to other endpoint too.

Reply:
Makes sense.